tests/disaster-recovery/mdr/test_app_failover_and_relocate_when_zone_down.py

```diff
@@ -0,0 +1,272 @@
+                and get_fence_state(self.primary_cluster_name) == "Fenced"
+            ):
+                enable_unfence(self.primary_cluster_name)
+                gracefully_reboot_ocp_nodes(self.namespace, self.primary_cluster_name)
+
+        request.addfinalizer(finalizer)
+
+    @pytest.mark.polarion_id("OCS-XXXX")
+    def test_application_failover_and_relocate(
+        self,
+        setup_acm_ui,
+        nodes_multicluster,
+        dr_workload,
+        node_restart_teardown,
+    ):
+
+        """
+        Tests to verify failover and relocate all apps in a single zone after a zone disruption
+
+        """
+
+        if config.RUN.get("mdr_failover_via_ui"):
+            ocs_version = version.get_semantic_ocs_version_from_config()
+            if ocs_version <= version.VERSION_4_12:
+                logger.error(
+                    "ODF/ACM version isn't supported for Failover/Relocate operation"
+                )
+                raise NotImplementedError
+
+        acm_obj = AcmAddClusters()
+        workload = dr_workload(num_of_subscription=1)[0]
+        self.namespace = workload.workload_namespace
+
+        # Create application on Primary managed cluster
+        set_current_primary_cluster_context(workload.workload_namespace)
+        self.primary_cluster_name = get_current_primary_cluster_name(
+            namespace=workload.workload_namespace
+        )
+
+        # ToDo: To create backupschedule in active hub and verify the backups are taken
+
+        # Get nodes from zone where active hub running
+        config.switch_ctx(get_active_acm_index())
+        active_hub_index = config.cur_index
+        zone = config.ENV_DATA.get("zone")
+        active_hub_cluster_node_objs = get_node_objs()
+        set_current_primary_cluster_context(workload.workload_namespace)
+        if config.ENV_DATA.get("zone") == zone:
+            managed_cluster_index = config.cur_index
+            managed_cluster_node_objs = get_node_objs()
+        else:
+            set_current_secondary_cluster_context(workload.workload_namespace)
+            managed_cluster_index = config.cur_index
+            managed_cluster_node_objs = get_node_objs()
+        external_cluster_node_roles = config.EXTERNAL_MODE.get(
+            "external_cluster_node_roles"
+        )
+        ceph_node_ips = []
+        for ceph_node in external_cluster_node_roles:
+            if (
+                external_cluster_node_roles[ceph_node].get("location").get("datacenter")
+                != "zone-b"
+            ):
+                continue
+            else:
+                ceph_node_ips.append(
+                    external_cluster_node_roles[ceph_node].get("ip_address")
+                )
+
+        # Shutdown one zones
+        logger.info("Stopping all nodes from one zone b")
+        nodes_multicluster[managed_cluster_index].stop_nodes(managed_cluster_node_objs)
+        nodes_multicluster[active_hub_index].stop_nodes(active_hub_cluster_node_objs)
```

Comment:
Can have a dedicated function for shutting down all clusters in a zone..by passing context as args.

Reply:
I will take this with another PR, along with application set creation